lib/protocol/events.js

```diff
@@ -227,17 +228,18 @@ function eventsProtocolFactory (
             Constants.Protocol.Exchanges.Events.Name,
             'graph.started' + '.' + graphid,
             function(data) {
-                callback(data.status);
+                callback(data);
             }
         );
     };
 
-    EventsProtocol.prototype.publishGraphFinished = function (graphid, status, nodeId) {
+    EventsProtocol.prototype.publishGraphFinished = function (graphid, data, nodeId) {
         var self = this;
         assert.uuid(graphid);
-        assert.string(status);
-
-        var data = { status: status };
+        assert.object(data);
+        assert.uuid(data.graphId);
```

Comment:
Don't need to check data's members here. it's adding limitation to this function, you could add checking in consuming place